contracts/carrot-app/src/handlers/instantiate.rs

```diff
@@ -22,61 +15,33 @@ pub fn instantiate_handler(
     app: App,
     msg: AppInstantiateMsg,
 ) -> AppResult {
-    let pool: Pool = PoolmanagerQuerier::new(&deps.querier)
-        .pool(msg.pool_id)?
-        .pool
-        .unwrap()
-        .try_into()?;
+    // We check the balance strategy is valid
+    msg.balance_strategy.check()?;
+
+    // We don't check the dex on instantiation
```

Comment:
If there are queries we need for the app to work it's best to also do them here. That ensures we catch missing entries early.

Reply:
These are now done inside the check function.